pkg/instance/cache.go

```diff
@@ -0,0 +1,90 @@
+package instance
+
+import (
+	"encoding/json"
+	"sync"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/cozy/cozy-stack/pkg/config"
+	"github.com/go-redis/redis"
+)
+
+var cacheExpiration = 5 * time.Minute
+
+type instanceCache interface {
+	Get(string) *Instance
+	Set(string, *Instance)
+	Revoke(string)
+}
+
+var _ instanceCache = &noCache{}
+var _ instanceCache = &redisCache{}
+
+type noCache struct{}
+
+func (*noCache) Get(string) *Instance  { return nil }
+func (*noCache) Set(string, *Instance) {}
+func (*noCache) Revoke(string)         {}
+
+type redisCache struct {
+	mu     sync.RWMutex
```

Comment:
Do we really need a mutex? I'd have think that the redis client have a pool of connections and be goroutine-safe.